e2e-tests/tests/basic-happy-path.spec.ts

```diff
@@ -50,29 +49,26 @@ describe('Basic Happy Path', () => {
       }
     });
     if (allTestsSucceeded || Cypress.env('REMOVE_APP_ON_FAIL')) {
-      // use UI to remove the application to test the flow
-      // The below command aims to navigate to applications page.
-      // but it does not work well. Because when I add the
-      // 'GetAppStartedPage.waitForLoad()' after the step, it failed.
+      // Use UI to remove the application to test the delete flow
       Common.navigateTo(NavItem.applications);
-      // we only delete the app when cy get the app.
-      // it means we would skip 'delete app' sometimes.
-      length = Cypress.$(`[data-id="${applicationName}"`).length;
-
-      if (length > 0) {
-        Applications.openKebabMenu(applicationName);
-        cy.get(actions.deleteApp)
-          .its('length')
-          .then((deleteLength) => {
-            if (deleteLength > 0) {
-              cy.get(actions.deleteApp).click();
-              cy.get(actions.deleteModalInput).clear().type(applicationName);
-              cy.get(actions.deleteModalButton).click();
-            }
-          });
-      }
 
-      APIHelper.deleteGitHubRepository(repoOwner, repoName);
+      // Wait for the application row to be visible before attempting deletion
+      cy.get(`[data-id="${applicationName}"]`, { timeout: 30000 })
```

Comment:
Is there a reason for a 30-second timeout? The 'Common.navigateTo' already waits for all loaders from the page to disappear, so the page should be ready when searching for the app, right?